service/controller/resource/promremotewrite/create.go

```diff
@@ -0,0 +1,75 @@
+package promremotewrite
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/v7/pkg/controller/context/resourcecanceledcontext"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	r.logger.Debugf(ctx, "reconcile remotewrite")
+	{
+		// get remotewrite
+		remoteWrite, err := ToRemoteWrite(obj)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		selector, err := metav1.LabelSelectorAsSelector(&remoteWrite.Spec.ClusterSelector)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		// fetch current prometheus
```

Comment:
Please elaborate a bit, with something like :

```suggestion
		// fetch current prometheus using the selector provided in remoteWrite resource.
```